internal/engine/wazevo/backend/isa/arm64/instr.go

```diff
@@ -212,7 +213,11 @@ const (
 	useKindAMode
 	useKindRNAMode
 	useKindCond
-	useKindVecRRRRewrite
+	// useKindRDRewrite indicates an instruction where RD is used both as a source and destination.
```

Comment:
atomic CAS has the same register behavior as vec rewrite. I tried to find a name and comment to clarify this but happy to change as needed